deploy/twingate-operator/tests/deployment_test.yaml

```diff
@@ -84,3 +84,8 @@ tests:
     asserts:
       - failedTemplate:
           errorPattern: "twingateOperator: Must validate one and only one schema (oneOf)"
+  - it: should use optional values
+    values:
+      - values.test.yaml
+    asserts:
+      - matchSnapshot: {}
```

Comment:
Why don't we test optional values like the other tests in this file? I find this snapshot test quite problematic
- The `values.test.yaml` still have a mixed of default values and customized values. It's confusing to other maintainers what should go into that file.
- When I add a required values, I wouldn't expect to update this snapshot.

Reply:
Initially, I thought we might want to test other customized values like `podLabels`, `podAnnotations`, `nodeSelector`, etc which would make the test bloated. On second thought, i think we don't need to test these field since it's trivial. 

So we can just keep the test simple and only test `logFormat`, `logVerbosity`, `defaultResourceTags` and `extraEnvVars` which have some custom logic in `deployment.yaml`